src/output.c

```diff
@@ -239,6 +263,34 @@ static void output_id3(output_t *st, unsigned int program, uint8_t *buf, unsigne
                           price, until, url, seller, desc, received_as);
             }
         }
+        else if (memcmp(tag, "COMM", 4) == 0)
+        {
+            if (frame_len < 5)
+            {
+                log_warn("bad COMM tag (frame_len %d)", frame_len);
+            }
+            else
+            {
+                uint8_t enc = data[0];
+                uint8_t *delim = id3_strlen(enc, data + 4, frame_len - 4);
+                uint8_t *end = data + frame_len;
+
+                if (delim)
+                {
+                    nrsc5_id3_comment_t* prev = comm;
+
+                    comm = calloc(1, sizeof(nrsc5_id3_comment_t));
+                    comm->lang = strndup((char*) data + 1, 3);
+                    comm->short_content = id3_encode_utf8(enc, data + 4, delim - data - 4);
+                    comm->actual_text = id3_encode_utf8(enc, delim + 1, end - delim);
```

Comment:
Two bugs here:

1. The starting point for the string is incorrect in the UCS-2 case; it should be `delim + 2`.
2. The length calculation is incorrect because it doesn't subtract off the length of the delimiter itself (1 or 2). As a result, this reads past the end of the data buffer.

Perhaps it would be simpler to change `id3_strlen` to something like `id3_next_string` and have it return a pointer to the first byte after the delimiter.

Reply:
On second thought, that approach would just move the problem and make it harder to calculate the length of `short_content`.